src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -187,6 +206,151 @@ protected override IEnumerable<Model.AzureSqlManagedInstanceModel> GetEntity()
+
+            bool isHardwareFamilyChanged = !string.IsNullOrWhiteSpace(requestedHardwareFamily) && !currentHardwareFamily.Equals(requestedHardwareFamily, StringComparison.InvariantCultureIgnoreCase);
+
+            if (isHardwareFamilyChanged)
+            {
+                // Check whether managed instance is in instance pool because it is not possible to change hardware family in that case.
+                if (!string.IsNullOrWhiteSpace(existingInstance.InstancePoolName))
+                {
+                    throw new PSArgumentException(
+                        string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.InstancePoolInstanceCannotChangeHardwareFamily, this.Name, existingInstance.InstancePoolName),
+                        "ComputeGeneration");
+                }
+
+                // Check whether hardware family is being changed to a deprecated hardware family
+                if (requestedHardwareFamily.Equals(Constants.ComputeGenerationGen4, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    throw new PSArgumentException(
+                        Microsoft.Azure.Commands.Sql.Properties.Resources.NotPossibleToSwitchBackToGen4,
+                        "ComputeGeneration");
+                }
+
+                shouldConfirmHardwareFamilyChange = true;
+            }
+        }
+
+        /// <summary>
+        /// Get hardware family from SKU name.
+        /// </summary>
+        /// <returns>The model to send to the update</returns>
+        protected string GetHardwareGenerationFromSkuName(string skuName)
+        {
+            if (string.IsNullOrWhiteSpace(skuName))
+            {
+                return null;
+            }
+
+            string hardwareFamily = null;
+
+            if (skuName.Equals(Constants.GeneralPurposeGen4, StringComparison.InvariantCultureIgnoreCase)
+                || skuName.Equals(Constants.BusinessCriticalGen4, StringComparison.InvariantCultureIgnoreCase))
+            {
+                hardwareFamily = Constants.ComputeGenerationGen4;
+            }
+            else if (skuName.Equals(Constants.GeneralPurposeGen5, StringComparison.InvariantCultureIgnoreCase)
+                || skuName.Equals(Constants.BusinessCriticalGen5, StringComparison.InvariantCultureIgnoreCase))
+            {
+                hardwareFamily = Constants.ComputeGenerationGen5;
+            }
+
+            return hardwareFamily;
+        }
+
+        /// <summary>
+        /// Get edition from SKU name.
+        /// </summary>
+        /// <returns>The model to send to the update</returns>
+        protected string GetEditionFromSkuName(string skuName)
+        {
+            if (string.IsNullOrWhiteSpace(skuName))
+            {
+                return null;
+            }
+
+            string edition = null;
+
+            if (skuName.Equals(Constants.GeneralPurposeGen4, StringComparison.InvariantCultureIgnoreCase)
```

Comment:
This hardcoding of sku names / edition names on client side will add maintenance cost every time a new edition / gen is released. I think we need a different strategy here. For example, in CLI code we use capabilities API to find SKU based on requested parameters. Alternatively, perhaps you can reuse some of the existing edition/sku handling logic from db/elastic pool.

Reply:
Not needed any more since SkuName parameter is removed.